api/src/client/ApiFormationClient.test.ts

```diff
@@ -1470,6 +1469,55 @@ describe("ApiFormationClient", () => {
       });
     });
 
+    describe("Main Business Address when starting a new business", () => {
+      it("updates post body with undefined value when address is empty", async () => {
+        const stubResponse = generateApiResponse({});
+        mockAxios.post.mockResolvedValue({ data: stubResponse });
+
+        const userData = generateFormationUserData(
+          { businessPersona: "STARTING" },
+          {},
+          { addressLine1: "", addressMunicipality: undefined, addressZipCode: "" }
+        );
+
+        await client.form(userData, "some-url");
+        const postBody: ApiSubmission = mockAxios.post.mock.calls[0][1] as ApiSubmission;
+        expect(postBody.Formation.BusinessInformation.MainAddress).toBeUndefined();
+      });
```

Comment:
[sand] Can we add 1 more test to validate this logic doesn't get hit for Foreign businesses?

Reply:
done